extensions-core/avro-extensions/src/main/java/org/apache/druid/data/input/avro/GenericAvroJsonProvider.java

```diff
@@ -138,7 +141,9 @@ public void setArrayIndex(final Object o, final int i, final Object o1)
   @Override
   public Object getMapValue(final Object o, final String s)
   {
-    if (o instanceof GenericRecord) {
+    if (o == null) {
+      return null;
```

Comment:
Maybe `UNDEFINED` will be better

Reply:
Actually, using `UNDEFINED` puts it back into the state where it throws `PathNotFoundException` internally, in fixing the tests I lost the performance for json flattening. I've so far returned to using `null` but in a way that doesn't break json path filter expressions, but I'm still exploring to see if there is a more elegant way to handle this, maybe my own `NOT_FOUND` object to handle specially.

Reply:
> using `UNDEFINED` puts it back into the state where it throws `PathNotFoundException` internally
Okay, I got it. We should avoid creating `PathNotFoundException` instance, because it is expensive operation due to `java.lang.Throwable#fillInStackTrace()`.

> maybe my own `NOT_FOUND` object to handle specially
Sounds good to me. Otherwise, we should add the `@Nullable` annotation.

Reply:
> using `UNDEFINED` puts it back into the state where it throws `PathNotFoundException` internally

Okay, I understand. We should avoid creating a `PathNotFoundException` instance, because it is an expensive operation because of `java.lang.Throwable#fillInStackTrace()`. What I thought before is that if we return `UNDEFINED` in `getMapValue` method, the program will enter [this](https://github.com/json-path/JsonPath/blob/master/json-path/src/main/java/com/jayway/jsonpath/internal/path/PathToken.java#L45) `if` branch, but since we only  setup `Option.SUPPRESS_EXCEPTIONS` without `Option.REQUIRE_PROPERTIES`, we will still enter enter [this](https://github.com/json-path/JsonPath/blob/master/json-path/src/main/java/com/jayway/jsonpath/internal/path/PathToken.java#L64) branch to avoid creating a `PathNotFoundException`. However, my idea is wrong, it will cause the `EvaluationContextImpl#resultIndex` variable to be equals to zero, and enter [this](https://github.com/json-path/JsonPath/blob/master/json-path/src/main/java/com/jayway/jsonpath/internal/path/EvaluationContextImpl.java#L133) `if` branch, where a `PathNotFoundException` instance will inevitably be created.


> maybe my own `NOT_FOUND` object to handle specially

Sounds good to me. Otherwise, we should add the `@Nullable` annotation.

Reply:
I tried it out, and using a placeholder object is a bit more tedious than just using `null` but the code is more or less the same, so I think I will leave this PR as is. Do you think `@Nullable` important for methods that we will never likely call directly in the Druid codebase? I don't feel strongly either way and can add it if you or anyone else does, but in this case I don't really think it matters since we are only adjusting these methods to make json-path do the right thing.

Reply:
Right, I agree with you.